lib/ret/media_resolver.ex

```diff
@@ -191,31 +191,109 @@ defmodule Ret.MediaResolver do
     resolve_sketchfab_model(model_id, uri)
   end
 
-  defp resolve_non_video(%URI{} = uri, _root_host) do
-    # Fall back on og: tags
-    [uri, meta] =
-      case uri |> URI.to_string() |> retry_get_until_success([{"Range", "bytes=0-32768"}]) do
-        :error ->
-          nil
-
-        # note that there exist og:image:type and og:video:type tags we could use,
-        # but our OpenGraph library fails to parse them out.
-
-        # also, we could technically be correct to emit an "image/*" content type from the OG image case,
-        # but our client right now will be confused by that because some images need to turn into
-        # image-like views and some (GIFs) need to turn into video-like views.
-        resp ->
-          case resp.body |> OpenGraph.parse() do
-            %{video: video} when is_binary(video) ->
-              [URI.parse(video), %{expected_content_type: "video/*"}]
-
-            # don't send image/*
-            %{image: image} when is_binary(image) ->
-              [URI.parse(image), %{}]
-
-            _ ->
-              [uri, %{expected_content_type: content_type_from_headers(resp.headers)}]
+  defp resolve_non_video(%URI{host: host} = uri, _root_host) do
+    photomnemonic_endpoint = module_config(:photomnemonic_endpoint)
+
+    # Crawl og tags for hubs rooms + scenes
+    is_local_url = host === RetWeb.Endpoint.host()
+
+    # For text/html pages we use the screenshotter, otherwise return the raw URL.
+    # Try HEAD, if HEAD fails then do a GET and check content type.
+    # If HEAD works, check content type.
+    case uri |> URI.to_string() |> retry_head_until_success() do
```

Comment:
Control flow here feels pretty odd. Can't this case just always evaluate to either a valid http response or nil (Does the HEAD then GET fallback). And then do the content type check after to decide if it should pass to photomnemonic or do the og tags. Currently that second part happens in two places.